### ISSUE TITLE
FaceAnalysisDisplay.ByViewFacePointsAndValues node only works on single surface

### ISSUE THREAD

**Issue by Lexiko** [DynamoDS/Dynamo#6454]
## Dynamo version

0.9.2

## Operating system

Windows 7

## What did you do? 

Run FaceAnalysisDisplay.ByViewFacePointsAndValues and FaceAnalysisDisplay.ByViewSurfaceData

## What did you expect to see?

Coloured surfaces in Revit relating to analysis results

## What did you see instead?

Node outputs more than one Revit.Display and only one surface gets colorized in Revit
I've also started a thread here:
http://dynamobim.org/forums/topic/face-analysis-display-by-view-face-points-and-values-node-bug/
![archilab](https://cloud.githubusercontent.com/assets/18419461/14601235/6a8adf6c-05a4-11e6-8436-6234812c66e7.png)

**Comment by ikeough** [comment]
@Lexiko Sorry for the delay in reply. I think this might be a limitation of the Revit API which Dynamo is doing of good job of un-masking :( I have to check the API documentation, but I believe only one analysis display object is allowed in the view at one time, and Dynamo is creating an analysis display object for each surface. I need to investigate.

**Comment by Lexiko** [comment]
Yeah, that’s what my gut is telling me too. @ksobon has a node in archi-lab which creates a colored surface view(it seems to work with two surfaces: http://archi-lab.net/colored-surface-analysis-display-with-mantis-shrimp-and-ladybug-continued/). He uses it with mantis shrimp to bring in surfaces from grasshopper but I presume it should work the same with geometry out of revit. I tried to use it in my graph (see image in original post) but I keep getting a “null” result. I’m afraid I don’t know any programming syntax to figure it out myself :P

Alexis Kotzambasis

GroupGSA Pty Ltd
Level 7, 80 William St
East Sydney NSW
Australia 2011
D +612 8302 3196
T +612 9361 4144

ABN 76 002 113 779
nom architect M. Sheldon
ARN 3990 / ARBV 15637 / BAQ 3188
www.groupgsa.com<http://www.groupgsa.com>

[www.groupgsa.com]<http://www.groupgsa.com/>

**Comment by dimven** [comment]
Had a quick look at @ksobon 's implementation. The SpatialFieldManager allows for multiple surface references:
![2016-04-22_11-23-40](https://cloud.githubusercontent.com/assets/7148394/14730805/6f5328ee-087c-11e6-8f92-925d62119071.png)

![2016-04-22_11-14-15](https://cloud.githubusercontent.com/assets/7148394/14730806/71c47d80-087c-11e6-8f36-baa10c9b746e.png)

However Dynamo's implementation clears the "buffer" before adding each new surface:

https://github.com/ikeough/DynamoRevit/blob/Autogen/src/Libraries/RevitNodes/AnalysisDisplay/FaceAnalysisDisplay.cs#L45

Most likely it was done because it would be difficult to know if the newly added data is an update to an existing face reference or an entirely new instance?

**Comment by ksobon** [comment]
@dimven I can't believe you actually got my nodes to work. I thought they were just old and broken but at the same time I haven't used that particular node in a while. One issue I always had with these nodes was remapping of the UVs from Rhino UVs to Revit UVs as imported Rhino surfaces would get randomly flipped. Also, flat surfaces are a bit of a pickle for Revit when it comes to mapping UVs. For some reason it always kept the original domain of the surface rather then remapping it to 0-1 as it does with all other surfaces (nurbs). I am glad you got it working though.

**Comment by ParametricMonkey** [comment]
@ksobon This is a parallel topic. Any ideas of why this is failing and any suggestions on better workflows? 
https://github.com/ladybug-analysis-tools/ladybug-dynamo/issues/18#8 #
